### PR TITLE
Update index.rst #23732

### PR DESCRIPTION
**Pull request by DhavalParmar61**

### CODE REVIEW

**Comment by DhavalParmar61** [comment]
Removed broken link from the documentation as mentioned in issue #23732

File: doc/source/index.rst [hunk 1]
```diff
@@ -17,7 +17,6 @@ NumPy documentation
 **Version**: |version|
 
 **Download documentation**:
```
Comment (mattip) [inline doc/source/index.rst:19]: I think the "Download documentation" text is not needed anymore. What do you think? [this](https://output.circle-artifacts.com/output/job/df62f9e6-64b7-4c60-a4d4-9e49a3e83d0b/artifacts/0/doc/build/html/index.html) is what it looks like after rendering.
```suggestion
```

**Comment by mattip** [comment]
Thanks @DhavalParmar61

**Comment by DhavalParmar61** [comment]
@mattip The change is only reflected for devdocs but for documentation of other versions still the broken link is there.

**Pull request merged**